armi/physics/neutronics/latticePhysics/latticePhysicsWriter.py

```diff
@@ -241,11 +242,19 @@ def _getAllNuclidesByCategory(self, component=None):
         # on the components will already contain all the nuclides required to be
         # modeled by the lattice physics writer. Otherwise, assume that `allNuclidesInProblem`
         # should be modeled.
-        nuclides = (
-            sorted(objNuclides)
-            if self.explicitFissionProducts
-            else self.r.blueprints.allNuclidesInProblem
-        )
+        if self.explicitFissionProducts:
+            # If detailed axial expansion is active, mapping between blocks occurs on uniform mesh
+            # and this can cause blocks to have isotopes that they don't have cross sections for.
+            # Fix this by adding all isotopes so they are present it lattice physics.
+            if self.cs[CONF_DETAILED_AXIAL_EXPANSION]:
+                nuclides = self.r.blueprints.allNuclidesInProblem
+            else:
+                nuclides = sorted(objNuclides)
```

Comment:
I think we want to remove the `if` here and always use the `else`

Removing the dependency on `self.cs[CONF_DETAILED_AXIAL_EXPANSION]` for determining the set of nuclides to use is one of the purposes of this PR.

I think `objNuclides` will have the explicit fission products where we want in to (i.e., fuel) and not in other components.

Reply:
this is needed when you have a depletable block next to non-depletable so that all can have all nuclides.

Reply:
So `if self.explicitFissionProducts` is true and `self.cs[CONF_DETAILED_AXIAL_EXPANSION]` is true, then `nuclides = self.r.blueprints.allNuclidesInProblem` which is going to include all of the detailed FPs.

So `nucsToForceInXsGen` isn't going to add any new isotopes to `nuclides` because it already has them all.

I thought the purpose of `nucsToForceInXsGen` was specifically for the case of `self.explicitFissionProducts == true` and `self.cs[CONF_DETAILED_AXIAL_EXPANSION] == true`, so that we could specify a limited set of nuclides to generate XS for with having to include all of the fission products.

Reply:
We talked offline; this makes sense to me in the context of the features from #1257 having been reverted in this PR.

Even if `self.r.blueprints.nucsToForceInXsGen` doesn't add any new nuclides here, it will be used later by a subclass of the `LatticePhysicsWriter`.